packages/react-test-renderer/src/ReactTestRenderer.js

```diff
@@ -200,8 +200,45 @@ const validWrapperTypes = new Set([
   ClassComponent,
   HostComponent,
   ForwardRef,
+  // Normally skipped, but used when there's more than one root child.
+  HostRoot,
 ]);
 
+function getChildren(parent: Fiber) {
+  const children = [];
+  const startingNode = parent;
+  let node: Fiber = startingNode;
+  if (node.child === null) {
+    return children;
+  }
+  node.child.return = node;
```

Comment:
Granted this is copy-pasted - but wondering if you know, why do we mutate the fiber instance to set it's return here?

Reply:
So, this is a pattern we use everywhere we traverse fibers deeply. It's necessary because each fiber has two copies. The `return` pointer points to the parent but we don't know which copy (current of work in progress). If we just naïvely follow "return" to go up, we might end up in a different tree copy from the previous update.

This is why every time we descend down, no matter when, we always overwrite the `.return` pointer to the parent in the tree we're working on. This should have no observable side effect because we do this on *every* deep traversal, and we never trust the chain of `return` pointers to give us the "current" tree.